docs/Authentication.md

```diff
@@ -33,6 +33,52 @@ let provider = MoyaProvider<YourAPI>(plugins: [CredentialsPlugin { target -> URL
 ])
 ```
 
+Bearer HTTP Auth
+----------------
+
+Another common method of authentication is by using an access token. Commonly
+this is a [JWT](https://jwt.io/introduction/), but it can take other forms as
+well. These requests are authorized by adding an HTTP header of the following
+form:
+
+```
+Authorization: Bearer <token>
+```
+
+To make adding these authorization headers easier, Moya comes with an
+`AccessTokenPlugin`. It can be added to your provider like this:
+
+```swift
+let token = "eyeAm.AJsoN.weBTOKen"
+let authPlugin = AccessTokenPlugin(token: token)
+let provider = MoyaProvider<YourAPI>(plugins: [authPlugin])
+```
+
+By default, `AccessTokenPlugin` will add the authorization header to requests to
+any `TargetType`. If you want to control this behavior, you can have your
+`TargetType` conform to `AccessTokenAuthorizable` and use `shouldAuthorize` to
+specify the behavior:
```

Comment:
`shouldAuthorize` is implemented in the same way as `TargetType.validate` or `TargetType.parameters`. Normally you'd implement it as a switch on the `enum`. In my trivial test case, I made a `struct` that conforms to `TargetType` and `AccessTokenAuthorizable` and it takes an initialization parameter to indicate how is should respond to `shouldAuthenticate`. This is just to write more concise tests. This isn't the way you'd normally use it. In fact I think most people will be fine with the default behavior and not conform to `AccessTokenAuthorizable`. Does that make sense to you?